Add release notes for 0.12.2 (#798)
submariner-io/submariner-website: src/content/community/releases/_index.en.md

weight = 40
+++

## v0.12.2

This is a bugfix release:

> * Globalnet metrics port will now be opened by default when Globalnet is deployed.

Reviewer: ```suggestion
* Globalnet metrics port will now be opened by default when Globalnet is deployed using `subctl`.
```

> * Globalnet metrics port will now be opened by default when Globalnet is deployed using `subctl`.

Reviewer 2: ```suggestion
* Globalnet metrics ports will now be opened by default when Globalnet is deployed using `subctl`.
```

if there are several, or

```suggestion
* The Globalnet metrics port will now be opened by default when Globalnet is deployed using `subctl`.
```

if there is just one.

> * Globalnet metrics port will now be opened by default when Globalnet is deployed using `subctl`.

Reviewer 3: The current sentence talks about deployment and the first thought i get in my mind is its "subctl join ...". But we are talking about "subctl cloud prepare ...".

So, please clarify that metrics port is opened by default when cloud prepare is done via "subctl"

> * Submariner ServiceExport now has unique condition types to wait for readiness.

Reviewer 2: ```suggestion
* Submariner ServiceExport now has unique condition types to simplify waiting for readiness.
```

* The `subctl diagnose` command now supports NAT-discovery port validation.

> * The `cloud prepare` command will now work properly for nodes to which security groups were added manually.

Reviewer: ```suggestion
* The `subctl cloud prepare` command will now work properly for nodes to which security groups were added manually.
```

Reviewer 4: I think it should be 'subctl cloud prepare'

Reviewer: Oops, sorry!

> * The `subctl cloud prepare` command will now work properly for nodes to which security groups were added manually.

Reviewer 3: Since cloud prepare is very specific to the infrastructure on which its executed. Can you please clarify to which cloud (AWS/GCP/Azure/RHOS etc) this applies?

* The `submariner-operator` namespace is labeled in accordance with _KEP-2579: Pod Security Admission Control_ (default in Kubernetes 1.24)

> to allow the privileged Pods required for Submariner's interactions with iptables.

Reviewer 3: ```suggestion
to allow the privileged Submariner Pods.
```

IPtables interaction is just one of the things we do on the nodes. Apart from iptables, we also configure IPsec tunnels, routes, procs entries etc. So lets avoid mentioning iptables and just say that its for privileged submariner pods.

Reviewer 5: How about this wording?

```suggestion
to allow the Pods to be privileged.
```

Reviewer 6: I think this feedback was missed in the last revision.

* The default namespace for the `subctl diagnose` command is changed to `submariner-operator`.

> * Images are based on Fedora 36.

Reviewer 3: ```suggestion
* Submariner pod images are now based on Fedora 36.
```

Reviewer 7: Agree with Sridhar here,

Reviewer 6: I think this feedback was missed in the last revision.

> * Fix issues related to Globalnet pods not starting with OpenShift 4.11.

Reviewer 3: Can you please share the link of the issue/PR related to this note?

Author: issue: https://github.com/submariner-io/submariner/issues/1883
PR: https://github.com/submariner-io/submariner/pull/1895

Reviewer 3: Thanks. The issue was present in both Globalnet pods and route-agent pods and AFAICT is not specific to OCP. So can you please rephrase the sentence as follows.

"Fix issues related to Globalnet and Route-agent pods due to missing grep in the container image."

Reviewer 6: I think this feedback was missed in the last revision.
